pkg/controller/endpointslice/reconciler.go

```diff
@@ -50,6 +51,8 @@ type reconciler struct {
 	// topologyCache tracks the distribution of Nodes and endpoints across zones
 	// to enable TopologyAwareHints.
 	topologyCache *topologycache.TopologyCache
+	// eventRecorder allows reconciler to record and publish events.
+	eventRecorder record.EventRecorder
```

Comment:
I think you have to plumb it down from the controller here

https://github.com/kubernetes/kubernetes/blob/f8750e2faf99fb512c7c8fcce12ecaa8f793bd3c/pkg/controller/endpointslice/endpointslice_controller.go#L160-L166

Reply:
Right, thanks for catching this